web/modules/custom/asset_distribution/asset_distribution.module

```diff
@@ -70,3 +74,57 @@ function asset_distribution_rdf_load_bundle_alter($entity_values, &$bundle) {
     $bundle = \Drupal::entityTypeManager()->getStorage('rdf_type')->load('asset_release');
   }
 }
+
+
+/**
+ * Implements hook_entity_extra_field_info().
+ */
+function asset_distribution_entity_extra_field_info() {
+  $fields = [];
+
+  // Aggregate the licences of the distributions belonging to a solution in a
+  // pseudo-field.
+  $bundle = RdfEntityType::load('solution');
+  // During installation, this bundle might not yet be available.
+  if ($bundle) {
+    $fields['rdf_entity'][$bundle->id()]['display']['licence'] = [
+      'label' => t('Licence'),
+      'description' => t('The licence type, aggregated from the related releases'),
+      'visible' => TRUE,
+    ];
+  }
+
+  return $fields;
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_view().
+ */
+function asset_distribution_rdf_entity_view(array &$build, EntityInterface $entity, EntityViewDisplayInterface $display, $view_mode) {
+  if ($entity->bundle() !== 'solution' || !$display->getComponent('licence')) {
+    return;
+  }
+
+  // Retrieve all the distributions related to this solution.
+  $relation_manager = new AssetDistributionRelations(\Drupal::entityTypeManager());
```

Comment:
AssetDistributionRelations is a service. It is better to get it through 
\Drupal::service('asset_distribution.relations');